File: javascript/1582.Special-Positions-in-a-Binary-Matrix.js
/**
 * @param {number[][]} mat
 * @return {number}
 */
var numSpecial = function (mat) {
  let count = 0;
  const rows = mat.length;
  const cols = mat[0].length;
  const row_sum = Array(rows).fill(0);
  const col_sum = Array(cols).fill(0);

  for (let r = 0; r < rows; r++) {
    for (let c = 0; c < cols; c++) {
      if (mat[r][c] === 1) {
        row_sum[r] += 1;
        col_sum[c] += 1;
      }
    }
  }

  for (let r = 0; r < rows; r++) {
    for (let c = 0; c < cols; c++) {
      if (mat[r][c] === 1 && row_sum[r] === 1 && col_sum[c] === 1) {
        count += 1;
      }
    }
  }

  return count;
};

File: python/1582.Special-Positions-in-a-Binary-Matrix.py
class Solution:
    def numSpecial(self, mat: List[List[int]]) -> int:
        count = 0
        rows = len(mat)
        cols = len(mat[0])
        row_sum = [0] * rows
        col_sum = [0] * cols

        for r in range(rows):
            for c in range(cols):
                if mat[r][c] == 1:
                    row_sum[r] += 1
                    col_sum[c] += 1

        for r in range(rows):
            for c in range(cols):
                if mat[r][c] == 1 and row_sum[r] == 1 and col_sum[c] == 1:
                    count += 1

        return count